src/common/contexts.js

```diff
@@ -1,5 +1,9 @@
-import {createContext} from 'react';
+import { createContext } from "react";
 
-export const CredContext = createContext({ email: '', password: ''});
+export const CredContext = createContext([{ email: "", password: "" }]);
 
-export const AuthContext = createContext({ auth: false});
+export const UserContext = createContext({
+  email: "",
+  password: "",
+  auth: false,
+});
```

Comment:
You can probably split this module/page wise

Reply:
Done, I have created separate modules for each context alongside with their providers.